lib/Dialect/TritonGPU/Transforms/Pipeliner/SoftwarePipeliner.cpp

```diff
@@ -72,6 +72,9 @@ static bool pipelineLoop(scf::ForOp forOp, int numStages) {
   if (!preCondition(forOp))
     return false;
 
+  if (numStages < 2)
```

Comment:
should we remove the one line 118 then?

Reply:
Maybe not

Reply:
Oh, I get your point, you want the for loop's num_stage to overwrite the global setting. I agree then

Reply:
Good idea to do it more like a global control.